lib/array.js

```diff
@@ -1,47 +1,34 @@
 'use strict';
 
+const common = require('@metarhia/common');
+const { asyncIter } = require('./async-iterator.js');
+const { promisify } = require('util');
+
 // Asynchronous map (iterate parallel)
-//   items - <Array>, incoming
+//   items - <Iterable>, incoming
 //   fn - <Function>, to be executed for each value in the array
 //     current - <any>, current element being processed in the array
 //     callback - <Function>
 //       err - <Error> | <null>
 //       value - <any>
 //   done - <Function>, on done
 //     err - <Error> | <null>
-//     result - <Array>
-const map = (items, fn, done) => {
-  const len = items.length;
-  if (!len) {
-    done(null, []);
+//     result - <Iterable>
+const map = (items, fn, done = common.emptiness) => {
+  if (!items[Symbol.iterator]) {
```

Comment:
If I understand correctly, Iterable can also contain `Symbol.asyncIterator` and do not contain `Symbol.iterator`.
``` JavaScript
const asyncIterable = {
  [Symbol.asyncIterator]() {
    return {
      i: 0,
      next() {
        if (this.i < 3) {
          return Promise.resolve({ value: this.i++, done: false });
        }

        return Promise.resolve({ done: true });
      }
    };
  }
};

(async function() {
   for await (const i of asyncIterable) {
     console.log(i);
   }
})();
```

Reply:
AFAIU we can omit this check because `AsyncIterator` constructor will make them.